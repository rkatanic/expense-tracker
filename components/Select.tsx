interface Props {
  value: any;
  onChange: (e: any) => void;
  children: React.ReactNode;
  label?: string;
  name?: string;
}

const Select = ({
  value,
  onChange,
  children,
  label,
  name,
}: Props): JSX.Element => (
  <div>
    {label && (
      <label
        className="block mb-1 text-sm font-semibold text-zinc-900 dark:text-zinc-400"
        htmlFor={name}
      >
        {label}
      </label>
    )}
    <select
      data-testid="select"
      className="w-full text-zinc-900 rounded-md border bg-zinc-50 border-zinc-300 px-3 pr-9 py-1.5 text-sm focus:ring-emerald-500  focus:outline-0 dark:text-zinc-100 dark:bg-zinc-900 dark:border-zinc-500"
      value={value}
      onChange={onChange}
    >
      {children}
    </select>
  </div>
);

export default Select;
