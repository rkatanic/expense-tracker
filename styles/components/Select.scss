.select {
    font-size: 0.875rem;
    color: var(--color-white-one);
    outline: none;
    background-color: transparent;
    border: 0;
    border-bottom: 2px solid var(--color-gray-600);
    padding: 0.5rem 0;

    option {
        background-color: var(--color-gray-200);
        padding: 0.5rem 0;
    }
}