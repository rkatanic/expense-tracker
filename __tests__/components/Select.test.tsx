import { fireEvent, render } from "@testing-library/react";
import Select from "../../components/Select";

describe("Select", (): void => {
  it("should render", (): void => {
    const { baseElement } = render(
      <Select value="option one" onChange={jest.fn()}>
        <option>option one</option>
        <option>option two</option>
      </Select>
    );

    expect(baseElement).toMatchSnapshot();
  });

  it("should call onChange", (): void => {
    const mockOnChange = jest.fn();
    const { getByTestId } = render(
      <Select value="option one" onChange={mockOnChange}>
        <option>option one</option>
        <option>option two</option>
      </Select>
    );
    fireEvent.change(getByTestId("select"), {
      target: { value: "option one" },
    });

    expect(mockOnChange).toHaveBeenCalledTimes(1);
  });
});
