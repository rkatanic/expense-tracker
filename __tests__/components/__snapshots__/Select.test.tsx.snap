// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select should render 1`] = `
<body>
  <div>
    <div>
      <select
        class="w-full text-zinc-900 rounded-md border bg-zinc-50 border-zinc-300 px-3 pr-9 py-1.5 text-sm focus:ring-emerald-500 focus:outline-0 dark:text-zinc-100 dark:bg-zinc-900 dark:border-zinc-500"
        data-testid="select"
      >
        <option>
          option one
        </option>
        <option>
          option two
        </option>
      </select>
    </div>
  </div>
</body>
`;
